### PR TITLE
Fix: last error from stream_socket_enable_crypto

### LINKED ISSUE #120: Last error occurred not populated by stream_socket_enable_crypto
The PHP last error is not guaranteed to be populated by`stream_socket_enable_crypto`, so it can be `null` here https://github.com/WordPress/blueprints-library/blob/trunk/src/WordPress/AsyncHttp/Client.php#L491 if the function fails.

Should be changed to:

```php
$report = error_get_last();
$this->set_error( $request, new HttpError( 'Failed to enable crypto: ' . ( is_array( $report ) ? $report['message'] : 'unknown' ) ) );
```

### PR DESCRIPTION
**Pull request by zaerl**
### What does this PR do?
Fix: https://github.com/WordPress/blueprints-library/issues/120

### What problem does it fix?
See issue.

### How to test if it works?
Working unit tests is ok for this. `./vendor/bin/phpunit`

### CODE REVIEW

**Comment by brandonpayton** [comment]
This is interesting... we're getting the same unit test failures as you saw [here](https://github.com/WordPress/blueprints-library/pull/115#pullrequestreview-2491915686), @zaerl. I'm also not able to reproduce those locally.

From [the logs](https://github.com/WordPress/blueprints-library/actions/runs/12394440294/job/34806172151#step:5:11):
```
Run phpunit tests --testdox
  phpunit tests --testdox
  shell: /usr/bin/bash -e {0}
  env:
    LOCAL_PHP: 8.1-fpm
    PHPUNIT_CONFIG: phpunit.xml.dist
    COMPOSER_PROCESS_TIMEOUT: 0
    COMPOSER_NO_INTERACTION: 1
    COMPOSER_NO_AUDIT: 1
PHP Fatal error:  Uncaught Error: Call to undefined method PHPUnit\Util\Filesystem::classNameToFilename() in /home/runner/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php:495
Stack trace:
#0 /home/runner/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php(408): PHPUnit\TextUI\Command->handlePrinter()
#1 /home/runner/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php(114): PHPUnit\TextUI\Command->handleArguments()
#2 /home/runner/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php(99): PHPUnit\TextUI\Command->run()
#3 /home/runner/.composer/vendor/phpunit/phpunit/phpunit(107): PHPUnit\TextUI\Command::main()
#4 /home/runner/.composer/vendor/bin/phpunit(122): include('...')
#5 {main}

```

**Comment by zaerl** [comment]
> This is interesting... we're getting the same unit test failures as you saw [here](https://github.com/WordPress/blueprints-library/pull/115#pullrequestreview-2491915686), @zaerl. I'm also not able to reproduce those locally.
> 
> From [the logs](https://github.com/WordPress/blueprints-library/actions/runs/12394440294/job/34806172151#step:5:11):
> 
> ```
> Run phpunit tests --testdox
>   phpunit tests --testdox
>   shell: /usr/bin/bash -e {0}
>   env:
>     LOCAL_PHP: 8.1-fpm
>     PHPUNIT_CONFIG: phpunit.xml.dist
>     COMPOSER_PROCESS_TIMEOUT: 0
>     COMPOSER_NO_INTERACTION: 1
>     COMPOSER_NO_AUDIT: 1
> PHP Fatal error:  Uncaught Error: Call to undefined method PHPUnit\Util\Filesystem::classNameToFilename() in /home/runner/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php:495
> Stack trace:
> #0 /home/runner/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php(408): PHPUnit\TextUI\Command->handlePrinter()
> #1 /home/runner/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php(114): PHPUnit\TextUI\Command->handleArguments()
> #2 /home/runner/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php(99): PHPUnit\TextUI\Command->run()
> #3 /home/runner/.composer/vendor/phpunit/phpunit/phpunit(107): PHPUnit\TextUI\Command::main()
> #4 /home/runner/.composer/vendor/bin/phpunit(122): include('...')
> #5 {main}
> ```

I think is just a matter of phpunit version.

**Comment by brandonpayton** [comment]
> I think is just a matter of phpunit version.

@zaerl, this turned out to be a conflict between a global phpunit version and the deps composer installs under `./vendor`. I created a PR to explicitly use the `vendor/bin/phpunit`, and those errors went away.

#123

File: tests/unit/steps/UnzipStepRunnerTest.php [hunk 1]
```diff
@@ -73,9 +73,9 @@ public function testUnzipFileWhenUsingAbsolutePath() {
 		$extracted_file_path = $this->runtime->resolvePath( 'dir/test_zip.txt' );
 		$step->setExtractToPath( Path::getDirectory( $extracted_file_path ) );
 
+		$this->expectException(\Exception::class);
```
Comment (brandonpayton) [inline tests/unit/steps/UnzipStepRunnerTest.php]: Note:
Instead of expecting these exceptions which note an unimplemented, #123 marks each of the tests ask explicitly skipped because of the missing implementation and adds a TODOs to stop skipping once the step is implemented

**Review by brandonpayton — COMMENTED** [review]
The fix looks reasonable, but I don't have any experience with this code yet. Is there any place we are clearly testing or exercising the AsyncHttp Client?

**Comment by zaerl** [comment]
> The fix looks reasonable, but I don't have any experience with this code yet. Is there any place we are clearly testing or exercising the AsyncHttp Client?

It is crashing on local with fatal errors sometimes. I don't think we need a unit test for this, just be sure we do not treat a null as an array.

**Review by brandonpayton — APPROVED** [review]
This looks good to me. Thank you!
